### PR TITLE
Fix/tail capture segment class cast

### PR DESCRIPTION
**Pull request by vaslabs**
See https://github.com/http4s/rho/pull/598 but for 0.23

### CODE REVIEW

**Review by zarthross — APPROVED** [review]
Good work!   Normally I'd flag this a bin-incompat (even though mima disagrees), changing from String to Segment could break a bunch of stuff, but since the last release was RC1... I  think we can get away with sneaking this into Series/0.23.

**Comment by zarthross** [comment]
@FrancescoSerra Any thoughts here?   Let's make sure to forward port this into main.

File: core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala [hunk 1]
```diff
@@ -63,7 +63,9 @@ class RhoRoutesSuite extends CatsEffectSuite with RequestRunner {
         Ok("twoparams2_" + foo + bar.getOrElse("cat"))
     }
 
-    GET / "variadic" / * |>> { tail: List[String] => Ok("route8_" + tail.mkString("/")) }
+    //identity is used to force the type usage of the path tail
+    GET / "variadic" / * |>> { tail: List[Segment] =>
+      Ok("route8_" + tail.map(identity).mkString("/")) }
```
Comment (zarthross) [inline core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala:68]:
```suggestion
    //encoded is used to force the type usage of the path tail
    GET / "variadic" / * |>> { tail: List[Segment] =>
      Ok("route8_" + tail.map(_.encoded).mkString("/")) }
```

**Comment by vaslabs** [comment]
> Good work! Normally I'd flag this a bin-incompat (even though mima disagrees), changing from String to Segment could break a bunch of stuff, but since the last release was RC1... I think we can get away with sneaking this into Series/0.23.

I would agree but I thought that the binary incompatibility broke before this change, maybe with some http4s update? something that switched the tail type from string to segment, I didn't go that deep into the investigation. What I mean, this PR doesn't really introduce binary incompatibility

File: core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala (hunk 1 above)
Comment (FrancescoSerra) [reply core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala:68]: @vaslabs are you happy with incorporating @zarthross ' suggestion?

Comment (vaslabs) [reply core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala:68]: I thought I pushed it already but maybe I only did on the other branch

Comment (vaslabs) [reply core/src/test/scala/org/http4s/rho/RhoRoutesSuite.scala]: done

**Pull request merged**